libutils/json.c

```diff
@@ -1353,6 +1353,28 @@ void JsonArrayAppendElement(
     SeqAppend(array->container.children, element);
 }
 
+void JsonArrayExtend(JsonElement *a, JsonElement *b)
+{
+    assert(a != NULL);
+    assert(a->type == JSON_ELEMENT_TYPE_CONTAINER);
+    assert(a->container.type == JSON_CONTAINER_TYPE_ARRAY);
+    assert(b != NULL);
+    assert(b->type == JSON_ELEMENT_TYPE_CONTAINER);
+    assert(b->container.type == JSON_CONTAINER_TYPE_ARRAY);
+
+    size_t len = JsonLength(b);
+    for (size_t i = 0; i < len; i++)
+    {
+        JsonArrayAppendElement(a, JsonAt(b, i));
```

Comment:
I thought this could go one step further and use the knowledge of the `JsonElement` internals and only do `SeqAppendSeq()` on the sequences storing the child elements.